### PR TITLE
Add ddoc fullText to the CLI switches

### PR DESCRIPTION
**Pull request by wilzbach**
Required for automatically generating the CLI options at dlang.org, see https://github.com/dlang/dlang.org/pull/2068
There were the following differences:

- `-gc` (only on dlang.org - the flag has been removed)
- `profile` (not mentioned on dlang.org)
- `-run` (not mentioned at the DMD output)
- `--version` (not mentioned on dlang.org)

The DMD CLI wasn't alphabetical ordered for a few cases (https://dlang.org/dmd.html is).
(in the transition from the manual CLI output to the dynamically generated one, I kept the ordering to allow easy diffing of the output).

Generating the language transitions automatically depends on https://github.com/dlang/dmd/pull/7648
Anyways, the available language transitions aren't shown on dlang.org at the moment.

### CODE REVIEW

File: src/dmd/cli.d
Comment (wilzbach) [inline src/dmd/cli.d]: I'm not so sure that the CLI ages are the best place to document this. Imho we should just link to https://dlang.org/spec/betterc.html and improve this page if needed.

File: src/dmd/cli.d
Comment (wilzbach) [inline src/dmd/cli.d]: That's the text from dlang.org - imho a bit more descriptive.

File: src/dmd/cli.d
Comment (wilzbach) [inline src/dmd/cli.d]: I don't think that the paragraphs are needed here, but it's a 1:1 from dlang.org

File: src/dmd/cli.d
Comment (wilzbach) [inline src/dmd/cli.d]: The "inline expand" looks out of place. I would just say "Inline functions"

File: src/dmd/cli.d
Comment (jacob-carlborg) [inline src/dmd/cli.d]: Ideally all these parameters should be documented.

**Comment by JinShil** [comment]
I'm a little skeptical about this documentation being in the binary unless it actually serves some for users of the compiler.  At the moment I think this is only used as a datastore to keep all related information about the compiler flags in one place.  That's perfectly fine, but I think we should version it out (may actually need string mixins) so it is only used for generating the documentation, but doesn't end up in the published binary.

**Comment by wilzbach** [comment]
> I'm a little skeptical about this documentation being in the binary unless it actually serves some for users of the compiler. 

It's needed for dlang/dlang.org#2068

> but doesn't end up in the published binary.

That's why the CLI text is generated at CTFE, so these strings shouldn't end up in the binary.
I made this more explicit with `version(DdocOptions)` as I checked the binary and DMD didn't seem to be smart enough. It would be cool to avoid the need for `version` though, but apparently DMD judges a template instantiation at CTFE reson enough to send a templated function to object code.

File: src/dmd/cli.d
Comment (wilzbach) [reply src/dmd/cli.d]: Done.

**Comment by WalterBright** [comment]
> That's why the CLI text is generated at CTFE, so these strings shouldn't end up in the binary.
I made this more explicit with version(DdocOptions) as I checked the binary and DMD didn't seem to be smart enough. It would be cool to avoid the need for version though, but apparently DMD judges a template instantiation at CTFE reson enough to send a templated function to object code.

Please file a bugzilla for this. But it still needs to be versioned out so it doesn't appear in the dmd binary, because updating the dmd that builds dmd may be a looong wait.

**Comment by wilzbach** [comment]
> Please file a bugzilla for this.

https://issues.dlang.org/show_bug.cgi?id=18238

**Comment by wilzbach** [comment]
So as this is now using `version(DdocOptions)` and won't add up to the executable size. Is everyone OK with moving forward?
As mentioned the main benefit here is that there will be only _one_ source of truth for the compiler arguments and dlang.org/dmd.html will automatically be in sync.

**Comment by JinShil** [comment]
> Is everyone OK with moving forward?

LGTM

I don't see the output in DAutoTest.  Please advise.

**Comment by wilzbach** [comment]
> I don't see the output in DAutoTest. Please advise.

Yes, this PR is a dependency of https://github.com/dlang/dlang.org/pull/2068
I could probably change https://github.com/dlang/dlang.org/pull/2068 to use `static if` to softly fail if no ddoxText attribute is available, but unfortunately one of the two PRs has to be merged without the preview and I thought this PR here is easier to review, because it just adds text.

**Review by JinShil — APPROVED** [review]

File: src/dmd/cli.d
Comment (marler8997) [inline src/dmd/cli.d]: Maybe pull this method out of the version blocks since it appears to be the same in both cases?

File: src/dmd/cli.d
Comment (marler8997) [inline src/dmd/cli.d]: Maybe
```
version(DdocOptions) this.ddocText = ddocText;
```
Then remove the other instance of this constructor?

**Review by marler8997 — APPROVED** [review]

**Comment by JinShil** [comment]
@wilzbach This has been open for a week, which I think is ample time for others to weigh in.  I'll probably merge this if you address @marler8997's comments above and no other objections arise.

**Comment by wilzbach** [comment]
Cool! I will do so once I am back and have access to my laptop.

**Comment by wilzbach** [comment]
(Rebased, addressed the review and resolved the merge conflict.)

File: src/dmd/cli.d
Comment (JinShil) [reply src/dmd/cli.d]: I agree, but let's leave that to another PR.

File: src/dmd/cli.d
Comment (JinShil) [reply src/dmd/cli.d]: I agree.  Sorry, I didn't comment about  that earlier.  Care to make the change to "Inline functions..." before we merge?

**Pull request merged**
